src/socks-proxy/tests/test_smtp.py

```
@@ -0,0 +1,66 @@
+    
+    # Wait for message and get connection info
+    time.sleep(1)
+    messages = mock_smtp_server.get_messages()
+    assert len(messages) == 1, "Message should be received"
+    assert messages[0]["subject"] == "Test Direct Connection"
+    
+    # Log connection info for debugging
+    connection_info = messages[0].get("connection_info", {})
+    logger.info(f"Direct SMTP connection info: {connection_info}")
+
+    assert connection_info["peer_host"] == "127.0.0.1"
```

Ruff (0.12.2)

Use of `assert` detected

(S101)